denops/@ddc-sources/buffer.ts

```diff
@@ -59,7 +62,9 @@ export class Source extends BaseSource<Params> {
     denops: Denops,
     bufnr: number,
   ): Promise<Candidate[]> {
-    return allWords(await fn.getbufline(denops, bufnr, 1, "$"))
+    const iskeyword = await op.iskeyword.getLocal(denops);
+    const pattern = "[" + vimoption2ts(iskeyword) + "]*";
```

Comment:
I am not sure that we should use `sourceOptions.keywordPattern`, which is the pattern for the current edited buffer only.

Imagine we are editing a Clojure file and we have in split windows a Java file or an SQL file or a Markdown file, each buffer has it's own iskeyword option that make sense in their file type. Do we want to collect words of the other buffers with the Clojure ([rather wide](https://github.com/clojure-vim/clojure.vim/blob/b54a4d6a01950398330090683d5b304cc8a5f71f/ftplugin/clojure.vim#L20)) iskeyword ?

Reply:
`iskeyword` from `op.iskeyword.getLocal()` is also local to the current edited buffer, so collecting words with inappropriate `iskeyword` can't be avoided.

I think this is not so big problem because ddc-buffer rarely gathers words from other buffers .
`makeCurrentBufCache()`, which gathers words from other buffers, is called when there is no cache for the buffers to gather words, and this happens only on initialization.

Reply:
Ok, I understand. Will update the PR accordingly.

Reply:
Hi Matsui54,

I just found out that we can get the `iskeyword` buffer local value using denops api `fn.getbufvar`.

I have tested it successfully, by opening a markdown and a typescript file in the same tab and then enter insert mode to trigger onInit and onEvent methods. Here are the logs, tracing events and `iskeyword` resolution (buffer 4 is the typescript file which has a longer pattern).

![2022-01-19_ddc-buffer_test-iskeyword](https://user-images.githubusercontent.com/1496827/150033849-b3b4a679-0557-4e8d-bb1a-94b64c4955a6.png)

Would that be a better implementation?

Reply:
Great! But at the same time, we should consider `sourceOptions.keywordPattern` as possible because users may want to override `iskeyword` by this ddc-option.
How about doing like this?
```typescript
    let pattern;
    if (bufnr == (await fn.bufnr(denops))) {
      // current buffer
      pattern = sourceOptions.keywordPattern;
    } else {
      // other buffer
      const iskeyword = await fn.getbufvar(denops, bufnr, "&iskeyword") as string
      pattern = "[" + vimoption2ts(iskeyword) + "]*";
    }
```

Reply:
Ok, I will try to do something in this way.